lib/buffer.js

```diff
@@ -807,7 +807,7 @@ Buffer.prototype.toJSON = function() {
 
 
 function adjustOffset(offset, length) {
-  offset = +offset;
+  offset |= 0;
   if (offset === 0 || Number.isNaN(offset)) {
```

Comment:
The `Number.isNaN(offset)` portion of this conditional should be able to be removed with this change since `|` will cause invalid inputs to always be zero (whereas `+` will not).